docs/tcpAgent.md

```diff
@@ -0,0 +1,97 @@
+```
+Capture the result to pass into the agent invocation.
+
+You can also capture the value by visiting the Instance Identity page, at something like "https://myjenkins.example.com/instance-identity". If you use this mechanism you must stitch it back together into a single, complete value from its multi-line display representation.
+
+This mechanism requires a download of the `agent.jar`, as described for "Download JNLP file".
+
+Once all the prerequisite files and data have been obtained, the agent can be launched with a command like this
+```
+java -cp agent.jar hudson.remoting.jnlp.Main \
+  -headless \
+  -workDir <work directory> \
+  -direct <HOST:PORT> \
+  -protocols JNLP4-connect \
+  -instanceIdentity <instance identity> \
+  <secretString> <agentName>
+```
+The "-protocols" parameter is optional, but is useful to limit the agent to protocols the server supports. The only currently supported and recommended protocol is "JNLP4-connect".
+
+Note that this mechanism uses a different JAR entry point than used for "Download JNLP file". The parameters available and the default behavior may be different between the entry points.
+
+### Install as Windows service
+On a Microsoft Windows platform, you can install the agent as a Windows Service. This allows the Windows Service infrastructure to manage the process lifecycle.
+
+To configure the agent this way, first launch the agent using one of the other mechanisms that doesn't run as headless. In the agent GUI, select "File -> Install as a service". Additional descriptions of configuring this mechanism are located at [Installing Jenkins as a Windows service](https://wiki.jenkins.io/display/JENKINS/Installing+Jenkins+as+a+Windows+service) or [How to Install Several Windows Slaves as a Service?](https://support.cloudbees.com/hc/en-us/articles/217423827-How-to-Install-Several-Windows-Slaves-as-a-Service-)
+
+## Parameters
+
+There are a number of different launch parameters that control how the agent connects and behaves. The parameters available and the default behavior may vary depending upon the entry point. You can obtain usage information by executing `java -cp agent.jar hudson.remoting.jnlp.Main` or `java -jar agent.jar --help`. Not all parameters work together and some parameters require the use of others.
+
+There are also system or environment variables that control some advanded behaviors documented at [Remoting Configuration](https://github.com/jenkinsci/remoting/blob/master/docs/configuration.md). Many of these need to be set at both master and agent. Changing some of these may result in unreliable behavior. One of these parameters that is supported with defined behavior is [`NO_PROXY`](https://github.com/jenkinsci/remoting/blob/master/docs/no_proxy.md). Similar to the usage in a number of other applications, this controls which hosts should be excluded from proxy configurations.
+
+### The '@' argument annotation
```

Comment:
And also add some documentation on the '@' argument annotation that has been missing.